app/models/user.rb

```diff
@@ -200,6 +202,17 @@ def name=(name)
     self.full_name = name
   end
 
+  def self.from_omniauth(auth)
+    includes(:user_identities)
+      .where(
+        'user_identities.provider=? AND user_identities.uid=?',
+        auth.provider,
+        auth.uid
+      )
+      .references(:user_identities)
+      .first
```

Comment:
Very *nitpicky*, but would `.take` here work? Because it's faster than `.first`.